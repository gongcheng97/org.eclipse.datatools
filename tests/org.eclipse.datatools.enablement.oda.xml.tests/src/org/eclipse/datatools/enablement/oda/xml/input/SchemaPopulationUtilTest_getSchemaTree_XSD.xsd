<?xml version="1.0" encoding="UTF-8"?>
<xs:schema
    xmlns:xs="http://www.w3.org/2001/XMLSchema"
    elementFormDefault="qualified"
    attributeFormDefault="unqualified"
   >
 
    <!-- ABSTRACT TYPES -->
    <xs:complexType name="NamedElement" abstract="true">
        <xs:attribute name="name"/>
        <xs:attribute name="fqn"/>
    </xs:complexType>
    
    <xs:complexType name="Classifier" abstract="true">
        <xs:complexContent>
            <xs:extension base="NamedElement">
                <xs:sequence minOccurs="0" maxOccurs="unbounded">
                    <xs:element name="attribute" type="Attribute" minOccurs="0" maxOccurs="unbounded"/>
                    <xs:element name="operation" type="Operation" minOccurs="0" maxOccurs="unbounded"/>
                    <xs:element name="class" type="Class" minOccurs="0" maxOccurs="unbounded"/>
                </xs:sequence>
            </xs:extension>
        </xs:complexContent>
    </xs:complexType>
    

    
    <!-- CONCRETE TYPES -->
    <xs:complexType name="Attribute">
        <xs:complexContent>
            <xs:extension base="NamedElement"/>
        </xs:complexContent>
    </xs:complexType>
    
    <xs:complexType name="Operation">
        <xs:complexContent>
            <xs:extension base="NamedElement"/>
        </xs:complexContent>
    </xs:complexType>
    
    <xs:complexType name="Class" >
        <xs:complexContent>
            <xs:extension base="Classifier"> 
 			<xs:sequence minOccurs="0" maxOccurs="unbounded">
                    <xs:element name="class" type="Class" minOccurs="0" maxOccurs="unbounded"/>
               	</xs:sequence>
		</xs:extension>
        </xs:complexContent>
    </xs:complexType>
    
    <xs:complexType name="Interface">
        <xs:complexContent>
            <xs:extension base="Classifier"/> 
        </xs:complexContent>
    </xs:complexType>
    
    <xs:complexType name="Diagram">
        <xs:attribute name="imagePath" use="required"/>
    </xs:complexType>
    
    <xs:complexType name="Package">
        <xs:complexContent>
            <xs:extension base="NamedElement">
                <xs:sequence minOccurs="0" maxOccurs="unbounded">
                    <xs:element name="package" type="Package" minOccurs="0" maxOccurs="unbounded"/>
                    <xs:element name="class" type="Class" minOccurs="0" maxOccurs="unbounded"/>
                    <xs:element name="interface" type="Interface" minOccurs="0" maxOccurs="unbounded"/>
                    <xs:element name="diagram" type="Diagram" minOccurs="0" maxOccurs="unbounded"/>
                </xs:sequence>
            </xs:extension>
        </xs:complexContent>
    </xs:complexType>

    
    
     <xs:element name="model" type="Package"/>
    
    
</xs:schema>