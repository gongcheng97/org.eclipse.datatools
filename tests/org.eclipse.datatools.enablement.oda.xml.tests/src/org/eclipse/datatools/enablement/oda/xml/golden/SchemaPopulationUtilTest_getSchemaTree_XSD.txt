  ROOT:0:0
    model:1:1
      package:1:2
        package:1:3
          package:1:4
            package:1:5
              package:1:6
                package:1:7
                  package:1:8
                    package:1:9
                      package:1:10
                      class:1:10
                      interface:1:10
                      diagram:1:10
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    interface:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    diagram:1:9
                      imagePath:2:10
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  interface:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  diagram:1:8
                    imagePath:2:9
                  name:2:8
                  fqn:2:8
                class:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                interface:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                diagram:1:7
                  imagePath:2:8
                name:2:7
                fqn:2:7
              class:1:6
                attribute:1:7
                  name:2:8
                  fqn:2:8
                operation:1:7
                  name:2:8
                  fqn:2:8
                class:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                name:2:7
                fqn:2:7
              interface:1:6
                attribute:1:7
                  name:2:8
                  fqn:2:8
                operation:1:7
                  name:2:8
                  fqn:2:8
                class:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                name:2:7
                fqn:2:7
              diagram:1:6
                imagePath:2:7
              name:2:6
              fqn:2:6
            class:1:5
              attribute:1:6
                name:2:7
                fqn:2:7
              operation:1:6
                name:2:7
                fqn:2:7
              class:1:6
                attribute:1:7
                  name:2:8
                  fqn:2:8
                operation:1:7
                  name:2:8
                  fqn:2:8
                class:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                name:2:7
                fqn:2:7
              name:2:6
              fqn:2:6
            interface:1:5
              attribute:1:6
                name:2:7
                fqn:2:7
              operation:1:6
                name:2:7
                fqn:2:7
              class:1:6
                attribute:1:7
                  name:2:8
                  fqn:2:8
                operation:1:7
                  name:2:8
                  fqn:2:8
                class:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                name:2:7
                fqn:2:7
              name:2:6
              fqn:2:6
            diagram:1:5
              imagePath:2:6
            name:2:5
            fqn:2:5
          class:1:4
            attribute:1:5
              name:2:6
              fqn:2:6
            operation:1:5
              name:2:6
              fqn:2:6
            class:1:5
              attribute:1:6
                name:2:7
                fqn:2:7
              operation:1:6
                name:2:7
                fqn:2:7
              class:1:6
                attribute:1:7
                  name:2:8
                  fqn:2:8
                operation:1:7
                  name:2:8
                  fqn:2:8
                class:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                name:2:7
                fqn:2:7
              name:2:6
              fqn:2:6
            name:2:5
            fqn:2:5
          interface:1:4
            attribute:1:5
              name:2:6
              fqn:2:6
            operation:1:5
              name:2:6
              fqn:2:6
            class:1:5
              attribute:1:6
                name:2:7
                fqn:2:7
              operation:1:6
                name:2:7
                fqn:2:7
              class:1:6
                attribute:1:7
                  name:2:8
                  fqn:2:8
                operation:1:7
                  name:2:8
                  fqn:2:8
                class:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                name:2:7
                fqn:2:7
              name:2:6
              fqn:2:6
            name:2:5
            fqn:2:5
          diagram:1:4
            imagePath:2:5
          name:2:4
          fqn:2:4
        class:1:3
          attribute:1:4
            name:2:5
            fqn:2:5
          operation:1:4
            name:2:5
            fqn:2:5
          class:1:4
            attribute:1:5
              name:2:6
              fqn:2:6
            operation:1:5
              name:2:6
              fqn:2:6
            class:1:5
              attribute:1:6
                name:2:7
                fqn:2:7
              operation:1:6
                name:2:7
                fqn:2:7
              class:1:6
                attribute:1:7
                  name:2:8
                  fqn:2:8
                operation:1:7
                  name:2:8
                  fqn:2:8
                class:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                name:2:7
                fqn:2:7
              name:2:6
              fqn:2:6
            name:2:5
            fqn:2:5
          name:2:4
          fqn:2:4
        interface:1:3
          attribute:1:4
            name:2:5
            fqn:2:5
          operation:1:4
            name:2:5
            fqn:2:5
          class:1:4
            attribute:1:5
              name:2:6
              fqn:2:6
            operation:1:5
              name:2:6
              fqn:2:6
            class:1:5
              attribute:1:6
                name:2:7
                fqn:2:7
              operation:1:6
                name:2:7
                fqn:2:7
              class:1:6
                attribute:1:7
                  name:2:8
                  fqn:2:8
                operation:1:7
                  name:2:8
                  fqn:2:8
                class:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                name:2:7
                fqn:2:7
              name:2:6
              fqn:2:6
            name:2:5
            fqn:2:5
          name:2:4
          fqn:2:4
        diagram:1:3
          imagePath:2:4
        name:2:3
        fqn:2:3
      class:1:2
        attribute:1:3
          name:2:4
          fqn:2:4
        operation:1:3
          name:2:4
          fqn:2:4
        class:1:3
          attribute:1:4
            name:2:5
            fqn:2:5
          operation:1:4
            name:2:5
            fqn:2:5
          class:1:4
            attribute:1:5
              name:2:6
              fqn:2:6
            operation:1:5
              name:2:6
              fqn:2:6
            class:1:5
              attribute:1:6
                name:2:7
                fqn:2:7
              operation:1:6
                name:2:7
                fqn:2:7
              class:1:6
                attribute:1:7
                  name:2:8
                  fqn:2:8
                operation:1:7
                  name:2:8
                  fqn:2:8
                class:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                name:2:7
                fqn:2:7
              name:2:6
              fqn:2:6
            name:2:5
            fqn:2:5
          name:2:4
          fqn:2:4
        name:2:3
        fqn:2:3
      interface:1:2
        attribute:1:3
          name:2:4
          fqn:2:4
        operation:1:3
          name:2:4
          fqn:2:4
        class:1:3
          attribute:1:4
            name:2:5
            fqn:2:5
          operation:1:4
            name:2:5
            fqn:2:5
          class:1:4
            attribute:1:5
              name:2:6
              fqn:2:6
            operation:1:5
              name:2:6
              fqn:2:6
            class:1:5
              attribute:1:6
                name:2:7
                fqn:2:7
              operation:1:6
                name:2:7
                fqn:2:7
              class:1:6
                attribute:1:7
                  name:2:8
                  fqn:2:8
                operation:1:7
                  name:2:8
                  fqn:2:8
                class:1:7
                  attribute:1:8
                    name:2:9
                    fqn:2:9
                  operation:1:8
                    name:2:9
                    fqn:2:9
                  class:1:8
                    attribute:1:9
                      name:2:10
                      fqn:2:10
                    operation:1:9
                      name:2:10
                      fqn:2:10
                    class:1:9
                      attribute:1:10
                      operation:1:10
                      class:1:10
                      name:2:10
                      fqn:2:10
                    name:2:9
                    fqn:2:9
                  name:2:8
                  fqn:2:8
                name:2:7
                fqn:2:7
              name:2:6
              fqn:2:6
            name:2:5
            fqn:2:5
          name:2:4
          fqn:2:4
        name:2:3
        fqn:2:3
      diagram:1:2
        imagePath:2:3
      name:2:2
      fqn:2:2
